filebeat/crawler/registrar.go

```diff
@@ -212,28 +199,15 @@ func (r *Registrar) getPreviousFile(newFilePath string, newFileInfo os.FileInfo)
 }
 
 func (r *Registrar) setState(path string, state *FileState) {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
 	r.state[path] = state
 }
 
 func (r *Registrar) getState(path string) (*FileState, bool) {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
 	state, exist := r.state[path]
 	return state, exist
 }
 
-func (r *Registrar) updateStateSource(state *FileState, path string) {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
-	state.Source = path
-}
-
 func (r *Registrar) getStateCopy() map[string]FileState {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
-
 	copy := make(map[string]FileState)
 	for k, v := range r.state {
 		copy[k] = *v
```

Comment:
The worst case here would be that the FileStateOS is updated during the state is read, means a race condition. In this case, just an "old" file state would be persisted and next time the "up-to-date" filestate would be persisted. Any other potential side effects?